File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/QuitGame_Request.java
package org.emulinker.kaillera.controller.v086.protocol;

import com.google.auto.value.AutoValue;
import org.emulinker.kaillera.controller.messaging.MessageFormatException;

@AutoValue
public abstract class QuitGame_Request extends QuitGame {
  private static final String DESC = "Quit Game Request";
  private static final String USERNAME = "";
  private static final int USER_ID = 0xFFFF;

  public static AutoValue_QuitGame_Request create(int messageNumber) throws MessageFormatException {
    V086Message.validateMessageNumber(messageNumber, DESC);

    return new AutoValue_QuitGame_Request(messageNumber, ID, DESC, USERNAME, USER_ID);
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/QuitGame_Request.kt
package org.emulinker.kaillera.controller.v086.protocol

import org.emulinker.kaillera.controller.messaging.MessageFormatException
import org.emulinker.kaillera.controller.v086.protocol.V086Message.Companion.validateMessageNumber

data class QuitGame_Request
    @Throws(MessageFormatException::class)
    constructor(override val messageNumber: Int) : QuitGame() {

  override val username = ""
  override val userId = 0xFFFF

  init {
    validateMessageNumber(messageNumber)
  }
}
